en/computer_vision/collision_prevention.md

```diff
@@ -196,10 +209,70 @@ make px4_sitl gz_x500_lidar_2d
 
 Next, adjust the relevant parameters to the appropriate values and add arbitrary obstacles to your simulation world to test the collision prevention functionality.
 
-The diagram below shows how the simulation looks when viewed in Gazebo.
+The diagram below shows a simulation of collision prevention as viewed in Gazebo.
 
 ![RViz image of collision detection using the x500_lidar_2d model in Gazebo](../../assets/simulation/gazebo/vehicles/x500_lidar_2d_viz.png)
 
-<!-- PR companion collision prevention (initial): https://github.com/PX4/PX4-Autopilot/pull/10785 -->
-<!-- PR for FC sensor collision prevention: https://github.com/PX4/PX4-Autopilot/pull/12179 -->
-<!-- using rangefinder? -->
+## Sensor Data Overview (Implementation Details)
+
+Collision Prevention has an internal obstacle distance map that divides the plane around the drone into 72 Sectors.
+Internally this information is stored in the [`obstacle_distance`](../msg_docs/ObstacleDistance.md) UORB topic.
+New sensor data is compared to the existing map, and used to update any sections that has changed.
+
+The angles in the `obstacle_distance` topic are defined as follows:
+
+![Obstacle_Distance Angles](../../assets/computer_vision/collision_prevention/obstacle_distance_def.svg)
+
+The data from rangefinders, rotary lidars, or companion computers, is processed differently, as described below.
+
+### Rotary Lidars
+
+Rotary Lidars add their data directly to the [`obstacle_distance`](../msg_docs/ObstacleDistance.md) uORB topic.
+
+### Rangefinders
+
+Rangefinders publish their data to the [`distance_sensor`](../msg_docs/DistanceSensor.md) uORB topic.
+
+This data is then mapped onto the `obstacle_distance` topic.
+All sectors which have any overlap with the orientation (`orientation` and `q`) of the rangefinder, and the horizontal field of view (`h_fov`) are assigned that measurement value.
+For example, a distance sensor measuring from 9.99° to 10.01° the measurements will get added to the bin's corresponding to 5° and 10° covering the arc from 2.5° and 12.5°
+
+::: info
+the quaternion `q` is only used if the `orientation` is set to `ROTATION_CUSTOM`.
+:::
+
+### Companion Computers
+
+Companion computers update the `obstacle_distance` topic using ROS2 or the [OBSTACLE_DISTANCE](https://mavlink.io/en/messages/common.html#OBSTACLE_DISTANCE) MAVLink message.
```

Comment:
@Claudio-Chies A rangefinder updated via a companion computer will publish distance sensor messages by default. Do the updates that come in via this path also get added to `obstacle_distance` or does the companion _explicitly_ have to update `obstacle_distance` ?

Reply:
Good question. From my understanding, any sensors connected to an onboard computer should be handled there and then published via an `obstacle_distance` message, because if we look at the topics available as an input via MicroXRCE then we only see  `/fmu/in/obstacle_distance `and no `/fmu/in/distance_sensor.`
the main use case would be if you have sensors like a depth camera which can not directly be integrated into PX4, you need the onboard computer to create a sensible obstacle_distance from this.

Now the collision_prevention does not differentiate between from where the data comes, if there is an onboarc computer which is able to publish distance_sensor messages with orientations other than down, then they will get merged into the obstacle map.
Does this answer the question?

Reply:
aaah, yes, it monitors the distance_sensor topic irrespective of the source, the only relevant thing is its orientation